vllm/core/scheduler.py

```diff
@@ -1583,6 +1583,43 @@ def _get_num_lookahead_slots(self, is_prefill: bool,
 
         return self.scheduler_config.num_lookahead_slots
 
+    def _get_budget_for_chunk(self, budget: SchedulingBudget) -> int:
+        """When doing chunked prefill, calculate the token budget for a single
+        chunk. This dynamically scales the chunk size down as the number of
+        sequences that require prefilling increases. This ensures that a single
+        sequence with a very large prompt to prefill doesn't take the entire
+        remaining token budget, allowing other sequences to prefill and decode
+        concurrently."""
```

Comment:
The statement "scales the chunk size down" is kind of confusing. IIUC, we don't actually reduce the chunk size for this batch, but reducing the "allocated token budget" for this request, so `_get_token_budget_for_request` may be a better naming.